### ISSUE TITLE
Connect To search not working with no letters entered

### ISSUE THREAD

**Issue by teleute** [scottdurow/SparkleXrm#98]
I've just installed the newest (release) versions of SparkleXrm and Refreshed Connections UI.  After I've put the web resource on a form, and I go to add a connection, I get the two fields - Connect To and Role.  If I just click the search magnifying glass next to Connect To, though, nothing happens.  It works fine when I put even a single letter in the box. though.  (Most of the time.  Every so often if I put something in the box and hit search, the first time it comes up with a red circle/X and says "Required".  Subsequent searches work, though.

Interestingly, the Role field works fine with nothing entered - it's only Connect To that doesn't.

I'm on 2016 on-premise, accessing from Win 10 through latest Chrome.  Tried IE as well and got the same result.

**Comment by scottdurow** [comment]
Do you have 7.2.9 of SparkleXRM and 1.1 of the Refresh Connection UI? 
I've tested this on a CRM2016 Org and it seems to be ok - could you use F12 to see if there are any JavaScript errors thrown?

**Comment by teleute** [comment]
Nope, no errors.  I'm on 7.2.8 though, as the wording in the download area implies that .9 is a pre-release (hence my "latest release" wording).  Do you think that the update might help?  I generally prefer to not run pre-release in a production environment...

**Comment by teleute** [comment]
Still hoping to get this sorted, so I tried the .9 pre-release version.  I have the same issue - no results if I search with no letters entered.  Still no javascript errors.  Tried in Chrome, FF, IE and Edge (all on Windows 10).  Also tried before and after the recent CRM service pack, since I needed to install it anyway, and it made no difference.

**Comment by teleute** [comment]
Have a bit more info:
In ClientUI.js, if I add an alert on the "term" variable at the start of RecordSearchCommand:    
```RecordSearchCommand: function(term, callback) {
alert(term);
```

On the first time I enter a search term after a reload, the alert is blank.  Subsequent times I search, any search term I put in that box comes back in the alert, and it properly returns results.  If I never put anything in the box as search term, the alert is always blank and it never returns any results.

I also tested this on a brand new entity with no other javascript or web resources, just to ensure I didn't have other code getting in the way.  Same behaviour.  :-(

Gonna call it a day for now, and start poking at it again tomorrow, but thought I'd post this much here for now in case it rang any bells for you in the meantime.  :-)  Thanks!

**Comment by scottdurow** [comment]
Good progress - thanks for looking at this.

**Comment by teleute** [comment]
Okay - I've partially tracked down the issue with it not showing any results with a blank entry.  If I set in the js to remove the `isquickfindfields="1"` parameter from the "or" filter, the blank entry query works just fine.  This doesn't solve the first search with a search entry problem, though, so it looks like they're separate.

Any ideas why isquickfindfields would cause an issue?  I'm just searching account and contact, and while we've made some modifications we haven't touched the main fields at all.

Thanks!

**Comment by teleute** [comment]
re: the other issue (the first search not working), I haven't been able to make any more progress.  I tried doing a stack trace on the js in Chrome, but just cannot track down where that value is going missing.  :-(  How does it get assigned from the input field to "term"?

**Comment by scottdurow** [comment]
Could you try and reproduce this in a CRM online trial and grant me access? I would really like to get to the bottom of it because I can't reproduce.

**Comment by teleute** [comment]
Unfortunately, last time I tried to open a CRM Online trial instance I went back and forth w/ MS for about a month and they still couldn't get one opened for me (we have an on-prem license through a charitable donation, so it majorly complicates things).  I can see if anything's changed on that front, but I'm not super hopeful.  :-(  Anything else I might be able to do to output useful debugging info in the meantime?

**Comment by danieldix12** [comment]
I'm also encountering both of these problems. The first I've found a resolution for but I'm still having trouble with the second one.

1) "Connect To" search not working:
I believe this issue is being caused by having columns other than text/lookup in the QuickFind view. See issue #95. Janis-veinbergs committed a fix to this on Apr 13 and once I stopped using the 1.1 managed solution and instead deployed from Visual Studio I no longer received this error.

2) First search not working:
I can consistently reproduce this issue in IE and Chrome by typing "Test" into record1id and then clicking out of the lookup field. Clicking out of the field triggers the validation rule added in ClientUI.ViewModel.ObservableConnection.validateRecord1Id. This then causes the Required error message to appear and clears out the text entered into the field. If I then type "Test" back into record1id and click out of the lookup field this does not trigger the validation rule, presumably because the error message is already displayed. If I click the magnifying glass button instead of clicking out of the lookup field, the validation rule is triggered before RoleSearchCommand, which is why alert(term) is returning NULL for teleute. I have not been able to determine why the validation rule is being triggered after clicking out of record1id.

![image](https://cloud.githubusercontent.com/assets/20250394/16751889/bf276a02-4791-11e6-9c6f-f44e6e20a19a.png)

**Comment by teleute** [comment]
D'oh - I missed that other issue thread because it wasn't the original issue I came on here to research/report.  :-)  I just happened to notice it whilst tracking down the first search not working issue.  Thanks!  So one (basically) down, one to go...  Sadly, the second one is the harder one to work around.
